### PR TITLE
[Source Tooling] Refactoring action to convert if statement to switch

### PR DESCRIPTION
**Pull request by Regno**
Implement action to convert if statements from:
  ```
var e = E.case1
  if e == .case1 {}
  else if e == .case2 {}
  else {}
```
to
```
  var e = E.case1
  switch e {
  case .case1: break
  case .case2: break
  default: break
  }
```
Resolves [SR-5740](https://bugs.swift.org/browse/SR-5740).

P.S. I can't reopen previous PR](https://github.com/apple/swift/pull/26719) because of force push to branch after rebase. I remade refactoring action to resolve comments from previous PR.

@nathawes please take a look to new implementation.

### CODE REVIEW

File: test/refactoring/ConvertToSwitchStmt/basic.swift [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+func checkEmptyBody(e: E) {
+  if e == .first {}
+  else if e == .second {
+    print("second")
+  }
+}
+
+// RUN: rm -rf %t.result && mkdir -p %t.result
+
+// RUN: %refactor -convert-to-switch-stmt -source-filename %s -pos=9:3 -end-pos=16:4 > %t.result/L9-3.swift
```
Comment (nathawes) [inline test/refactoring/ConvertToSwitchStmt/basic.swift]: It'd be good to type check the output file for each of these to make sure the code we're generating is valid and continues to type check as the compiler changes. You can do that be adding an extra RUN line like the below:
```
// RUN: %target-swift-frontend-typecheck %t.result/L9-3.swift
```

**Review by nathawes — APPROVED** [review]
This looks great! Thanks for all the improvements. Just left one suggestion to improve the tests, but I'm happy to merge this without it too.

**Comment by nathawes** [comment]
@swift-ci please test

**Comment by nathawes** [comment]
@swift-ci Please Sourcekit Stress test

**Comment by nathawes** [comment]
Looks like a few of the existing tests need updating to allow for the new refactoring kind being in the list:
```
/Users/buildnode/jenkins/workspace/swift-PR-osx/branch-master/swift/test/refactoring/RefactoringKind/basic.swift:431:30: error: CHECK-EXTRCT-METHOD-NEXT: is not on the line after the previous match
// CHECK-EXTRCT-METHOD-NEXT: Action ends
                             ^
<stdin>:4:1: note: 'next' match was here
Action ends
^
<stdin>:2:15: note: previous match ended here
Extract Method
              ^
<stdin>:3:1: note: non-matching line after previous match is here
Convert To Switch Statement
```

File: test/refactoring/ConvertToSwitchStmt/basic.swift (hunk 1 above)
Comment (Regno) [reply test/refactoring/ConvertToSwitchStmt/basic.swift]: Added for each test case.

**Comment by Regno** [comment]
Tests are failed at cases like this:
```
if a {
    return
  } else {
    return
  }
```

I think there are no sense to convert `if` statement with single bool condition to switch statement. So I just skip such kind of `if` at `isApplicable` function. Now all tests should be passed.

**Comment by futurejones** [comment]
Can one of the admins verify this patch?

**Comment by nathawes** [comment]
@swift-ci please test

**Pull request merged**